### PR TITLE
Azure: Settings for Azure AD Workload Identity

### PR DESCRIPTION
**Pull request by kostrse**
**What is this feature?**

This PR introduces new configuration settings for Azure AD Workload Identity authentication:
```ini
[azure]
workload_identity_enabled = true

# These settings override default Workload Identity and normally not required
workload_identity_tenant_id =
workload_identity_client_id =
workload_identity_token_file =
```

Details on Azure AD Workload Identity:
* https://azure.github.io/azure-workload-identity
* https://github.com/Azure/azure-workload-identity

**Why do we need this feature?**

Azure AD Workload Identity is the only supported way to pass a service identity to a Grafana instance hosted in Azure Kubernetes Service, or any Kubernetes cluster working with Azure Active Directory.

**Who is this feature for?**

Grafana users who host Grafana in AKS or standalone Kubernetes with AAD authentication.

**Which issue(s) does this PR fix?**:

Related  #75404

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: conf/sample.ini [hunk 1]
```diff
@@ -799,6 +799,18 @@
 # Should be set for user-assigned identity and should be empty for system-assigned identity
 ;managed_identity_client_id =
 
+# Azure AD Workload Identity
```
Comment (aangelisc) [inline conf/sample.ini:802]:
```suggestion
# Azure AD Workload Identity
# Specifies whether workload identity authentication should be enabled in datasources
# that support it. For more details on Workload Identity, review documentation here:
# https://azure.github.io/azure-workload-identity/docs/
```

Perhaps add this comment in all the `ini` files

**Comment by aangelisc** [comment]
I think in general this PR looks fine. Main needs to be merged in and the documentation for each new config field needs to be updated. Once that's done we can merge this and look at creating a PR for each datasource with more detailed documentation on the new authentication method.

File: conf/sample.ini (hunk 1 above)
Comment (kostrse) [reply conf/sample.ini]: Updated, please check wordings

**Review by aangelisc — APPROVED** [review]
This all LGTM, thank you @kostrse! I will merge this and implement in our datasources

**Pull request merged**